File: leetcode/1001-2000/1201-1300/1207-unique-number-of-occurrences/UniqueNumberOfOccurrences.ts
function uniqueOccurrences(numbers: number[]): boolean {
  const m = numbers.reduce(
    (accumulated, currentValue) => {
      accumulated[currentValue] = (accumulated[currentValue] ?? 0) + 1;

      return accumulated;
    },
    {}
  );

  const values = Object.values(m);
  const uniqueOccurrences = new Set(values);

  return values.length === uniqueOccurrences.size;
};

console.log(uniqueOccurrences([-3, 0, 1, -3, 1, 1, 1, -3, 10, 0]))

File: leetcode/1001-2000/1201-1300/1207-unique-number-of-occurrences/UniqueNumberOfOccurrences.java
import java.util.HashMap;
import java.util.HashSet;
import java.util.Map;
import java.util.Set;

public class UniqueNumberOfOccurrences {
  public static void main(String[] args) {
    int[] numbers = new int[] { -3, 0, 1, -3, 1, 1, 1, -3, 10, 0 };

    System.out.println(new Solution().uniqueOccurrences(numbers));
  }
}

class Solution {
  public boolean uniqueOccurrences(int[] numbers) {
    Map<Integer, Integer> m = new HashMap<>();

    for (int n : numbers) {
      m.merge(n, 1, Integer::sum);
    }

    Set<Integer> occurrences = new HashSet<>(m.values());

    return occurrences.size() == m.size();
  }
}
